src/main/java/backend/Logic.java

```diff
@@ -178,11 +178,25 @@ public String getDefaultRepo() {
         return models.getDefaultRepo();
     }
 
-    public CompletableFuture<Boolean> removeRepository(String repoId) {
-        models.removeRepoModelById(repoId);
+    public CompletableFuture<Boolean> removeStoredRepository(String repoId) {
         return repoIO.removeRepository(repoId);
     }
 
+    private void removeRepoModel(String repoId) {
+        models.removeRepoModelById(repoId);
+    }
+
+    public void removeUnusedModels(Set<String> reposInUse) {
+        // Not using getOpenRepositories since it returns all in lower case
+        // in which we can't feed to removeRepoModel(repoId)
+        Set<String> openRepos = models.toModels().stream().map(Model::getRepoId).collect(Collectors.toSet());
+        openRepos.removeAll(reposInUse);
```

Comment:
This can be written as a map, then filter, then forEach, since the set supports a fast `contains` operation.